### PR TITLE
add reverse k group LinkedList

### PR DESCRIPTION
**Pull request by BamaCharanChhandogi**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Comment by BamaCharanChhandogi** [comment]
@vie02 check!

File: src/test/java/com/thealgorithms/datastructures/lists/ReverseKGroupTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        head2.next.next.next = new Node(4);
+        head2.next.next.next.next = new Node(5);
+
+        Node result2 = reverser.reverseKGroup(head2, 3);
+        assertEquals(3, result2.value);
+        assertEquals(2, result2.next.value);
+        assertEquals(1, result2.next.next.value);
+        assertEquals(4, result2.next.next.next.value);
+        assertEquals(5, result2.next.next.next.next.value);
+        assertNull(result2.next.next.next.next.next);
```
Comment (siriak) [inline src/test/java/com/thealgorithms/datastructures/lists/ReverseKGroupTest.java:62]: Please extract this to a separate test

File: src/test/java/com/thealgorithms/datastructures/lists/ReverseKGroupTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        assertEquals(5, result.value);
+        assertNull(result.next);
+    }
+
+    @Test
+    public void testReverseKGroupWithMultipleElementsList() {
+        ReverseKGroup reverser = new ReverseKGroup();
+
+        // Create a list with multiple elements (1 -> 2 -> 3 -> 4 -> 5)
+        Node head = new Node(1);
```
Comment (siriak) [inline src/test/java/com/thealgorithms/datastructures/lists/ReverseKGroupTest.java:34]:
```suggestion
        Node head;
        head = new Node(1);
```
I'd write like that to conform to the pattern below

File: src/main/java/com/thealgorithms/datastructures/lists/ReverseKGroup.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package com.thealgorithms.datastructures.lists;
+
+/**
+ * Reverse K Group LinkedList
+ * Author: Bama Charan Chhandogi (https://github.com/BamaCharanChhandogi)
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/lists/ReverseKGroup.java:5]: Please add the problem statement or a link to it

**Review by siriak — APPROVED** [review]

**Pull request merged**